Reqnroll.Parser/CucumberMessages/CucumberMessagesConverter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Gherkin.CucumberMessages;
+using Gherkin.CucumberMessages.Types;
+
+namespace Reqnroll.Parser.CucumberMessages;
+
+/// <summary>
+/// Utility class that converts Reqnroll AST types in to 'CucumberMessages.Types' types.
+/// It uses two classes from the Gherkin project: AstMessagesConverter and PickleCompiler
+/// 
+/// Once the Gherkin project implementation directly emits CucumberMessages (eliminating the use of the Gherkin.CucumberMessages.Types namespace), this class can be removed
+/// </summary>
+public class CucumberMessagesConverter
+{
+    private readonly IIdGenerator _idGenerator;
+
+    public CucumberMessagesConverter(IIdGenerator idGenerator)
+    {
+        _idGenerator = idGenerator;
+    }
+
+    /// <summary>
+    /// This method transforms an AST <see cref="ReqnrollDocument"/> into a <see cref="GherkinDocument"/>.
+    /// Before doing so, it patches any missing Location elements in the AST. These might be missing because our ExternalData Plugin
+    /// does not emit Location elements for the Example table rows it generates.
+    /// </summary>
+    public GherkinDocument ConvertToCucumberMessagesGherkinDocument(ReqnrollDocument gherkinDocument)
+    {
+        var nullLocationPatcher = new PatchMissingLocationElementsTransformation();
+        var gherkinDocumentWithLocation = nullLocationPatcher.TransformDocument(gherkinDocument);
+        var converter = new AstMessagesConverter(_idGenerator);
+        var location = Path.Combine(gherkinDocument.DocumentLocation.FeatureFolderPath, Path.GetFileName(gherkinDocument.SourceFilePath));
```

Comment:
The `DocumentLocation.FeatureFolderPath` uses `/` as a folder separator independently of the OS, however the `Path.Combine` uses the path separator from the OS. 

We have the same problem a few lines below in the `ConvertToCucumberMessagesSource` method.

This results in a funny mixed output, like:

<img width="395" height="84" alt="image" src="https://github.com/user-attachments/assets/790c7be4-0b5c-490e-9b67-2368457adee5" />

I think we should use `/` everywhere in the cucumber messages and also in the HTML report, so I would replace `Path.Combine` with a custom "combine" method that uses `/`.